### DISCUSSION TITLE
Add support for inline Terraform variables

### DISCUSSION THREAD

**Discussion by stevehipwell** [aquasecurity/trivy#5142] [Ideas]
### Description

I'd like to be able to define Terraform variables in the config file (or as a CLI argument) instead of requiring a tfvars file.

The existing inputs could be reused with the presence of a `=` character changing the behaviour.

```shell
trivy config --tf-vars my_var=test
```

```yaml
terraform:
  vars:
    - my_var=test
```
    

### Target

None

### Scanner

None

**Comment by AnaisUrlichs** [comment]
How do you define terraform variables right now outside of trivy? hmm also with the trivy config file you would need to specify the variable in a .tfvars file instead of directly https://aquasecurity.github.io/trivy/v0.45/docs/references/configuration/config-file/#misconfiguration-options

I am not sure I follow what the issue is with that?

**Reply by stevehipwell** [reply]
@AnaisUrlichs this would be useful for reusable modules where the variables are undefined and tfvars file might be blocked.

**Reply by simar7** [reply]
> tfvars file might be blocked.

could you expand on this?

**Reply by stevehipwell** [reply]
I wouldn't expect to see a tfvars file in a re-usable module and generally block anyone accidentally adding one.

Inlining the tfvars in the Trivy configuration file would stop them being consumable by anything other than Trivy and make it 100% clear what their purpose is.
